api/src/mutations/mint.rs

```diff
@@ -495,6 +496,222 @@ impl Mutation {
         })
     }
 
+    pub async fn update_mint(
+        &self,
+        ctx: &Context<'_>,
+        input: UpdateMintInput,
+    ) -> Result<UpdateMintPayload> {
+        let AppContext {
+            db,
+            user_id,
+            organization_id,
+            balance,
+            ..
+        } = ctx.data::<AppContext>()?;
+        let credits = ctx.data::<CreditsClient<Actions>>()?;
+        let conn = db.get();
+        let solana = ctx.data::<Solana>()?;
+        let nft_storage = ctx.data::<NftStorageClient>()?;
+
+        let UserID(id) = user_id;
+        let OrganizationId(org) = organization_id;
+
+        let user_id = id.ok_or(Error::new("X-USER-ID header not found"))?;
+        let balance = balance
+            .0
+            .ok_or(Error::new("X-CREDIT-BALANCE header not found"))?;
+        let org_id = org.ok_or(Error::new("X-ORGANIZATION-ID header not found"))?;
+
+        let creators = input.creators;
+
+        let (mint, collection) = CollectionMints::find()
+            .find_also_related(Collections)
+            .filter(collection_mints::Column::Id.eq(input.mint))
+            .one(conn)
+            .await?
+            .ok_or(Error::new("Mint not found"))?;
+
+        if mint.creation_status != CreationStatus::Created {
+            return Err(Error::new("Mint not created"));
+        }
+
+        let collection = collection.ok_or(Error::new("Collection not found"))?;
+        let blockchain = collection.blockchain;
+
+        validate_creators(blockchain, &creators)?;
+        validate_json(blockchain, &input.metadata_json)?;
+
+        let seller_fee_basis_points = input.seller_fee_basis_points.unwrap_or_default();
+
+        let owner_address = fetch_owner(conn, collection.project_id, collection.blockchain).await?;
+
+        if collection.blockchain == BlockchainEnum::Solana {
+            validate_solana_creator_verification(&owner_address, &creators)?;
+        }
+
+        let creators_am = creators
+            .clone()
+            .into_iter()
+            .map(|creator| {
+                Ok(mint_creators::ActiveModel {
+                    collection_mint_id: Set(mint.id),
+                    address: Set(creator.address),
+                    verified: Set(creator.verified.unwrap_or_default()),
+                    share: Set(creator.share.try_into()?),
+                })
+            })
+            .collect::<Result<Vec<_>>>()?;
+
+        let deduction_id = credits
+            .submit_pending_deduction(
+                org_id,
+                user_id,
+                Actions::UpdateMint,
+                Blockchain::Solana,
```

Comment:
This should be the blockchain variable so if the action isn't supported the new version of the credit_client will reject the blockchain that don't support.